pinot-tools/src/main/java/org/apache/pinot/tools/admin/command/CreateSegmentCommand.java

```diff
@@ -228,15 +228,17 @@ public boolean execute()
       }
     }
 
-    FileFormat configFormat = segmentGeneratorConfig.getFormat();
-    if (_format == null) {
-      if (configFormat == null) {
-        throw new RuntimeException("Format cannot be null in config file.");
-      }
-      _format = configFormat;
-    } else {
-      if (configFormat != _format && configFormat != FileFormat.AVRO) {
-        LOGGER.warn("Find format conflict in command line and config file, use config in command line: {}", _format);
+    if (segmentGeneratorConfig.getRecordReaderPath() == null) {
```

Comment:
// Allow for instantiation general record readers from a record reader path passed into segment generator config
    // If this is set, this will override the file format
    if (recordReaderPath != null) {
      if (fileFormat != FileFormat.OTHER) {
        // NOTE: we currently have default file format set to AVRO inside segment generator config, do not want to break
        // this behavior for clients.
        LOGGER
            .warn("Using class: {} to read segment, ignoring configured file format: {}", recordReaderPath, fileFormat);
      }


Currently, in record reader factory, we have this default. Up to you, but potentially, we could follow the same model though requiring users to set file format to other seems annoying... maybe we can just leave a comment that the behavior is different or put it in relevant docs.

Reply:
Let me check the other path and keep them consistent if possible.